File: leave-service/src/main/java/com/otbs/leave/dto/LeaveResponse.java
package com.otbs.leave.dto;

import com.otbs.leave.model.ELeaveType;
import com.otbs.leave.model.EStatus;

import java.time.LocalDate;

public record LeaveResponse(
        Long id,
        String Name,
        String Department,
        LocalDate startDate,
        LocalDate endDate,
        ELeaveType leaveType,
        EStatus status
) {
}

File: app/src/main/java/com/example/onetechbs/db/LeaveResponse.kt
package com.example.onetechbs.db

import com.example.onetechbs.db.LeaveStatus
import com.example.onetechbs.db.ELeaveType

 import com.google.gson.annotations.SerializedName
import java.time.LocalDate

class LeaveResponse(
    val id: Long,
    @SerializedName("attachment") val attachment: String? = null,
    @SerializedName("Name") val name: String,
    @SerializedName("Department") val department: String,
    @SerializedName("startDate") val startDate: LocalDate,
    @SerializedName("endDate") val endDate: LocalDate,
    @SerializedName("leaveType") val leaveType: ELeaveType,
    @SerializedName("status") val status: EStatus
)